src/providers/delimitedtext/qgsdelimitedtextprovider.cpp

```diff
@@ -565,6 +597,14 @@ void QgsDelimitedTextProvider::scanFile( bool buildIndexes )
       mNumberFeatures++;
     }
 
+    // Progress changed every 100 features
+    if ( feedback && mNumberFeatures % 100 == 0 )
+    {
+      // We don't really know the progress because the number of records is not known
+      // but we need to signal to the caller that the scan is progressing
+      feedback->setProgress( mNumberFeatures );
```

Comment:
This is troublesome -- the api clearly defines the progress as 0-100, so setting it to other values is breaking that promise. 

If you need a way to indicate "records processed" then I'd suggest adding some other generic form of handling this to QgsFeedback (e.g. a "processedCountChanged" or "progressed" signal)